tests/Integration/Bridge/TolerantParser/SourceCodeFilesystem/ScfClassCompletorTest.php

```diff
@@ -81,16 +213,81 @@ public function provideComplete(): Generator
                     'type' => Suggestion::TYPE_CLASS,
                     'name' => 'Alphabet',
                     'short_description' => 'Test\Name\Alphabet',
+                    'class_import' => 'Test\Name\Alphabet',
+                ],
+                [
+                    'type' => Suggestion::TYPE_CLASS,
+                    'name' => 'Backwards',
+                    'short_description' => 'Test\Name\Backwards',
+                ],
+                [
+                    'type' => Suggestion::TYPE_CLASS,
+                    'name' => 'Clapping',
+                    'short_description' => 'Test\Name\Clapping',
+                ],
+                [
+                    'type' => Suggestion::TYPE_CLASS,
+                    'name' => 'WithoutNS',
+                    'short_description' => 'WithoutNS',
+                    'class_import' => null,
+                ],
+            ],
+        ];
+
+        yield 'new keyword with already imported class' => [
```

Comment:
Hmm, the `new` tests seem like they should be separete, i.e. just asserting that results are returned for `extends`, new`, `<a standalone classname>`, `use`, etc. would be less lines of code. in fact, I'm surprised there were not already tests for this :thinking: In anycase, repeating the same assertions probably adds more maintainance cost than value, especially seeing as they are all coupled to the same environment currently.